bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingAddOp.java

```diff
@@ -121,7 +121,7 @@ void sendWriteRequest(int bookieIndex) {
         int flags = isRecoveryAdd ? BookieProtocol.FLAG_RECOVERY_ADD : BookieProtocol.FLAG_NONE;
 
         lh.bk.getBookieClient().addEntry(lh.metadata.currentEnsemble.get(bookieIndex), lh.ledgerId, lh.ledgerKey,
-                entryId, toSend, this, bookieIndex, flags);
+                entryId, toSend, this, bookieIndex, flags, lh.writeFlags);
```

Comment:
isn't `writeFlags` a thing of `PendingAddOp`? if so, it should be passed in when constructing `PendingAddOps`s.

Reply:
We can save  a field, we have lh, there is no need to have a field in *PendingAddOp*. 
If you think it would be clearer I can add the field anyway

Reply:
doesn't we allow different flags for recovery_adds?

Reply:
done